Convert AnalysisOptions to Pydantic (Issue #78)

yongminkim0501 opened the pull request:
## Overview
This PR converts the 'AnalysisOptions' class to use Pydantic for more pythonic serialization, as suggested in #78 

## Changes
- Changed 'AnalysisOptions' from '@dataclass' to 'pydantic.BaseModel'
- Updated 'to_json()' to use '.dict()' for serialization (Pydantic v1 API)
- Simplified "from_json()' using Pydantic's dict unpacking ('**dict')
- Added "pydantic>=1.9.0, <1.10.0' to dependencies

## Technical Issue
- Used Pydantic v1.9 due to `typing-extensions` version conflict with `signal-processing-algorithms==1.3.5`
- Using '.dict()' instead of '.model_dump()' (v1.9 doesn't have model_dump, which is v2 only)

## Testing
- All 68 tests passing 

## Future work
- If this approach is approved, I plan to continue converting other classes with similar serialization patterns to Pydantic.

## Related Issue
Part of #78

henrikingo reviewed (commented):
Hi @yongminkim0501 and welcome!

As the person who wrote most of those to_json() methods, I'm glad to see them go away! 

I have just one small comment, if you can make that change, I'll merge this. And yes by all means, please continue this with remaining classes.

yongminkim0501 commented on otava/series.py:
```diff
@@ -32,26 +33,11 @@
-        self.min_magnitude = 0.0
-        self.orig_edivisive = False
-
-    def to_json(self):
```
Thanks for the feedback! I'll update the PR to deprecate the to_json() methods 
instead of removing them. Will commit the changes shortly.

yongminkim0501 committed:
Migrate to Pydantic and deprecate to_json() methods

- Convert data classes to Pydantic BaseModel
- Deprecate to_json() methods with DeprecationWarning
- Update all instantiations to use keyword arguments
- Add arbitrary_types_allowed config for nested models

Results: 11/12 tests passing

Note: test_orig_edivisive reveals a pre-existing issue where
compute_change_points_orig() returns EDivisive library objects
instead of ChangePoint objects. Could you help me understand
how to properly handle this case?

yongminkim0501 commented:
Thanks for the suggestion! I'll use pydantic.dataclass to make the tests pass while maintaining backward compatibility. 

Looking forward to discussing the preferred approach for data model management after the 0.7.0 release!

yongminkim0501 committed:
Refactor: Use pydantic.dataclasses to maintain backward compatibility

yongminkim0501 committed:
Merge branch 'master' into feature/pydantic-serialization

henrikingo commented:
> Apologies for the delayed response. I've been preoccupied with my university graduation project.

No need to apologize, this is a side project for all of us. A few days or weeks between interactions is normal.

> Thank you for the feedback! I've refactored the code to use `pydantic.dataclasses` as you suggested.
> 

Can you have a look at your own diff on the changes tab here. It looks like it now includes history that is a mix of different things: Some classes have moved to pydantic.dataclass, but actually the AnalysisOptions and some others are still  BaseModel. And some changes seem to just be formatting with no functional change.
